lib/iris/_data_manager.py

```diff
@@ -226,8 +226,8 @@ def data(self):
                         'before trying again.')
                 raise MemoryError(emsg.format(self.shape, self.dtype))
 
-        # Check the manager contract, as the managed data has changed.
-        self._assert_axioms()
+            # Check the manager contract, as the managed data has changed.
+            self._assert_axioms()
```

Comment:
I've indented this because we only need to assert the axioms if the state changed.